### PR TITLE
Trigger "forget word" by hotkey in pinyin and table engine

### PR DESCRIPTION
**Pull request by cinit**
Add a "forget word" option to the long-click menu of candidate words so that users can remove a candidate word without pressing Control+7.

image

### CODE REVIEW

**Comment by rocka** [comment]
Thank you for the contribution! This could serve as a good replacement implementation until we have real "candidate action" API in fcitx.

File: app/src/main/java/org/fcitx/fcitx5/android/input/candidates/CandidateItemUi.kt [hunk 1]
```diff
@@ -34,4 +41,48 @@ class CandidateItemUi(override val ctx: Context, theme: Theme) : Ui {
...
+
+    private var promptMenu: PopupMenu? = null
+
+    fun showExtraActionMenu(currentIm: InputMethodEntry, onForget: () -> Unit) {
+        promptMenu?.dismiss()
+        val context = root.context
+        promptMenu = PopupMenu(context, root).apply {
+            val actions = arrayListOf<Pair<String, () -> Unit>>().apply {
+                // only Pinyin Engine has forget word feature
+                if (currentIm.uniqueName == "pinyin" || currentIm.uniqueName == "shuangpin") {
```
Comment (rocka) [inline app/src/main/java/org/fcitx/fcitx5/android/input/candidates/CandidateItemUi.kt:53]: `table` based input methods also have the shortcut to "delete a word" in Ctrl+7, it's better to benfit them as well. I would add `addon` field to `InputMethodEntry` class later.

File: app/src/main/res/values-zh-rCN/strings.xml [hunk 2]
```diff
@@ -254,4 +254,6 @@
     <string name="table_dict_replace_message">码表词典 \"%1$s\" 存在。您可以导入新词典并替换它。</string>
     <string name="lang_switch_key_behavior">语言切换键行为</string>
     <string name="lang_switch_behavior_next_ime_app">切换至下一个输入法</string>
+    <string name="no_action_available">无可用操作</string>
+    <string name="action_forget_candidate_word">恢复默认词频</string>
```
Comment (rocka) [inline app/src/main/res/values-zh-rCN/strings.xml:258]: Please do not modify translations other than English manually. The translation is hosted on transifex: https://app.transifex.com/fcitx/fcitx5-android/translate/

**Review by rocka — CHANGES_REQUESTED** [review]

**Comment by mokapsing** [comment]
rime有个删除用户词的快捷键，是否有兴趣添加进去？
https://github.com/rime/librime/blob/1f3bf35691a5ad066ef36775e289525f3b092e28/src/rime/gear/editor.cc#L199
https://github.com/rime/librime/blob/1f3bf35691a5ad066ef36775e289525f3b092e28/src/rime/gear/editor.cc#L214

发送Ctrl+Delete键

**Review by rocka — APPROVED** [review]

**Pull request merged**